keps/sig-cluster-lifecycle/generic/1755-communicating-a-local-registry/README.md

```diff
@@ -0,0 +1,624 @@
+Include as much detail as possible so that people can understand the "how" of
+the system.  The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+Alice is setting up a development environment on Kubernetes.
+
+Alice creates a local Kind cluster. She follows the Kind-specific instructions
+for setting up a local registry during cluster creation.
+
+Alice will have some tool she interacts with that builds images and pushes them
+to a registry the cluster can access. That tool might be an IDE like VSCode,
+an infra-as-code toolchain like Pulumi, or a multi-service dev environment like Tilt.
+
+On startup, the tool should connect to the cluster and read the `local-image-loaders`
+ConfigMap. The ConfigMap embeds a versioned config.
+
+If the config specifies a registry location, the tool may automatically adjust
+image pushes to push to the specified registry.
+
+If the config specifies a `help` URL, the tool may prompt the user to set up a
+registry for faster development.
+
+### Notes/Constraints/Caveats (optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+- Users may see the ConfigMap and draw mistaken conclusions about how to
+  interact with it. They think deleting the ConfigMap would delete the local
+  registry (which it does not).
+
+- This KEP does not specify any automatic mechanism for keeping the
+  LocalRegistryStatus up-to-date with how the cluster is configured.
+  For example, the user might delete the registry. The cluster might
+  not have a way of knowing that the registry has died.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal and how do we mitigate.  Think broadly.
+For example, consider both security and how this will impact the larger
+kubernetes ecosystem.
+
+How will security be reviewed and by whom?
+
+How will UX be reviewed and by whom?
+
+Consider including folks that also work outside the SIG or subproject.
+-->
+
+None
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable.  This may include API specs (though not always
+required) or even code snippets.  If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### The `local-image-loaders` ConfigMap
+
+Tools that configure a local registry should apply a ConfigMap to the cluster.
+
+Documentation that educates people on how to set up a registry manually should
+include instructions on what the ConfigMap should be.
+
+The name of the ConfigMap must be `local-image-loaders`.
+
+The namespace must be `kube-public`.
+
+The fields of the ConfigMap have their field names formatted as
+`structureName.vX`.  `structureName` is a method for loading images into the
```

Comment:
but arguably `localRegistryStatus` does not imply a method for loading images. i think it requires a different name -potentially `localRegistryHosting`?

Reply:
ya, that seems fine to me